### ISSUE TITLE
`derive_partial_eq_without_eq` false positive on struct containing a `T: Fn()`

### ISSUE THREAD

**Issue by benwh1** [rust-lang/rust-clippy#9319]
### Summary

Haven't been able to trigger it when using any struct that doesn't contain some sort of function.

On stable, doesn't occur on 1.62.1, does occur on 1.63.0.

[Playground link](https://play.rust-lang.org/?version=beta&mode=debug&edition=2021&gist=d18231800c492b67abda23e9d258a409)

### Lint Name

derive_partial_eq_without_eq

### Reproducer

I tried this code:

```rust
#[derive(PartialEq, Eq)]
pub struct Foo<T: Fn()>(T);

fn main() {}
```

I saw this happen:

```
warning: you are deriving `PartialEq` and can implement `Eq`
 --> src/main.rs:1:10
  |
1 | #[derive(PartialEq, Eq)]
  |          ^^^^^^^^^ help: consider deriving `Eq` as well: `PartialEq, Eq`
  |
  = note: `#[warn(clippy::derive_partial_eq_without_eq)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#derive_partial_eq_without_eq
```

I expected to see this happen:

```
```

### Version

```text
rustc 1.65.0-nightly (29e4a9ee0 2022-08-10)
binary: rustc
commit-hash: 29e4a9ee0253cd39e552a77f51f11f9a5f1c41e6
commit-date: 2022-08-10
host: x86_64-unknown-linux-gnu
release: 1.65.0-nightly
LLVM version: 14.0.6
```

**Comment by J-ZhengLi** [comment]
This should be fixed by #12393 , somehow it did not get auto closed by bot
